fastplotlib/layouts/_gridplot.py

```diff
@@ -294,3 +308,94 @@ def __next__(self) -> Subplot:
+        self.maintain_aspect_button = ToggleButton(value=True, disabled=False, description="1:1",
+                                                   layout=Layout(width='auto'), tooltip='maintain aspect')
+        self.maintain_aspect_button.style.font_weight = "bold"
+        self.flip_camera_button = Button(value=False, disabled=False, icon='sync-alt',
+                                         layout=Layout(width='auto'), tooltip='flip')
+
+        positions = list(product(range(self.plot.shape[0]), range(self.plot.shape[1])))
+        values = list()
+        for pos in positions:
+            if self.plot[pos].name is not None:
+                values.append(self.plot[pos].name)
+            else:
+                values.append(str(pos))
+        self.dropdown = Dropdown(options=values, disabled=False, description='Subplots:')
+
+        self.widget = HBox([self.autoscale_button,
+                            self.center_scene_button,
+                            self.panzoom_controller_button,
+                            self.maintain_aspect_button,
+                            self.flip_camera_button,
+                            self.dropdown])
+
+        self.panzoom_controller_button.observe(self.panzoom_control, 'value')
+        self.autoscale_button.on_click(self.auto_scale)
+        self.center_scene_button.on_click(self.center_scene)
+        self.maintain_aspect_button.observe(self.maintain_aspect, 'value')
+        self.flip_camera_button.on_click(self.flip_camera)
+
+        self.plot.renderer.add_event_handler(self.click_handler, "click")
+
+    @property
+    def parser(self) -> Subplot:
+        # parses dropdown value as plot name or position
+        current = self.dropdown.value
+        if current[0] == "(":
+            return self.plot[eval(current)]
+        else:
+            return self.plot[current]
+
+    def auto_scale(self, obj):
+        current = self.parser
+        current.auto_scale(maintain_aspect=current.camera.maintain_aspect)
+
+    def center_scene(self, obj):
+        current = self.parser
+        current.center_scene()
+
+    def panzoom_control(self, obj):
+        current = self.parser
+        current.controller.enabled = self.panzoom_controller_button.value
+
+    def maintain_aspect(self, obj):
+        current = self.parser
+        current.camera.maintain_aspect = self.maintain_aspect_button.value
+
+    def flip_camera(self, obj):
+        current = self.parser
+        current.camera.scale.y = -1 * current.camera.scale.y
+
+    def click_handler(self, ev):
+        for subplot in self.plot:
+            pos = subplot.map_screen_to_world((ev.x, ev.y))
+            if pos is not None:
+                # update self.dropdown
+                if subplot.name is None:
+                    self.dropdown.value = str(subplot.position)
+                else:
+                    self.dropdown.value = subplot.name
+                subplot.controller.enabled = self.panzoom_controller_button.value
```

Comment:
You want it the other way around, click a Subplot sets the buttons from the Subplot values